Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -39,15 +39,22 @@ extension CallUIAdaptee {
         let callViewController = CallViewController(call: call)
         callViewController.modalTransitionStyle = .crossDissolve
 
-        guard let presentingViewController = UIApplication.shared.frontmostViewControllerIgnoringAlerts else {
+        guard let mainWindow = CurrentAppContext().mainWindow else {
+            owsFail("in \(#function) mainWindow unexpectedly nil")
+            return
+        }
+        guard let presentingViewController = mainWindow.rootViewController else {
             owsFail("in \(#function) view controller unexpectedly nil")
             return
         }
 
         if let presentedViewController = presentingViewController.presentedViewController {
-            presentedViewController.dismiss(animated: false)
+            presentedViewController.dismiss(animated: false) {
```

Comment:
This is a theory without proof/a repro.  I'm pretty sure that it's unsafe to assume `dismiss` finishes its work synchronously, even when animated is false, hence its completion handler.  So let's use the completion handler instead.